Libraries/LibWeb/HTML/AutocompleteElement.cpp

```diff
@@ -0,0 +1,369 @@
+        }
+        // Otherwise, let the element's autofill field name be "on".
+        else {
+            attr_details.field_name = "on"_string;
+        }
+
+        return attr_details;
+    };
+
+    // 1. If the element has no autocomplete attribute, then jump to the step labeled default.
+    if (!autocomplete_element_to_html_element().has_attribute(AttributeNames::autocomplete))
+        return step_default();
+
+    // 2. Let tokens be the result of splitting the attribute's value on ASCII whitespace.
+    auto tokens = autocomplete_tokens();
+
+    // 3. If tokens is empty, then jump to the step labeled default.
+    if (tokens.is_empty())
+        return step_default();
+
+    // 4. Let index be the index of the last token in tokens.
+    auto index = tokens.size() - 1;
+
+    // 5. Let field be the indexth token in tokens.
+    auto const& field = tokens[index];
+
+    // 6. Set the category, maximum tokens pair to the result of determining a field's category given field.
+    auto [category, maximum_tokens] = determine_a_field_category(field);
+
+    // 7. If category is null, then jump to the step labeled default.
+    if (!category.has_value())
+        return step_default();
+
+    // 8. If the number of tokens in tokens is greater than maximum tokens, then jump to the step labeled default.
+    if (tokens.size() > maximum_tokens.value())
+        return step_default();
+
+    // 9. If category is Off or Automatic but the element's autocomplete attribute is wearing the autofill anchor mantle,
+    //    then jump to the step labeled default.
+    if ((category == Category::Off || category == Category::Automatic)
+        && get_autofill_mantle() == AutofillMantle::Anchor)
+        return step_default();
+
+    // 10. If category is Off, let the element's autofill field name be the string "off", let its autofill hint set be empty,
+    //     and let its IDL-exposed autofill value be the string "off". Then, return.
+    if (category == Category::Off) {
+        attr_details.field_name = "off"_string;
+        attr_details.hint_set = {};
+        attr_details.value = "off"_string;
+        return attr_details;
+    }
+
+    // 11. If category is Automatic, let the element's autofill field name be the string "on", let its autofill hint set be empty,
+    //     and let its IDL-exposed autofill value be the string "on". Then, return.
+    if (category == Category::Automatic) {
+        attr_details.field_name = "on"_string;
+        attr_details.hint_set = {};
+        attr_details.value = "on"_string;
+        return attr_details;
+    }
+
+    // 12. Let scope tokens be an empty list.
+    Vector<String> scope_tokens;
+
+    // 13. Let hint tokens be an empty set.
+    Vector<String> hint_tokens;
+
+    // 14. Let credential type be null.
+    Optional<String> credential_type;
+
+    // 15. Let IDL value have the same value as field.
+    // NOTE: lowercasing is not mentioned in the spec, but required to pass all WPT tests.
```

Comment:
feels like a good question to ask in the HTML issue tracker.

Reply:
I posted on the [WHATWG matrix](https://matrix.to/#/!AGetWbsMpFPdSgUrbs:matrix.org/$5RmZTknw3nH_ZI5CXyHM1vvUuzyDgV1jxweW-Jp5oKY?via=matrix.org&via=mozilla.org&via=igalia.com)

Reply:
Filed an issue: https://github.com/whatwg/html/issues/11036